core/base/slot_chain.go

```diff
@@ -1,22 +1,27 @@
 package base
 
 import (
+	"sort"
 	"sync"
 
 	"github.com/alibaba/sentinel-golang/logging"
 	"github.com/alibaba/sentinel-golang/util"
 	"github.com/pkg/errors"
 )
 
-type NamedSlot interface {
+type BaseSlot interface {
 	// Name returns its slot name which should not be the same as other slots.
 	Name() string
+
+	// Order returns the order value of slot.
+	// Slots will be sorted according to order value in increasing order.
+	Order() uint32
 }
 
 // StatPrepareSlot is responsible for some preparation before statistic
 // For example: init structure and so on
 type StatPrepareSlot interface {
-	NamedSlot
+	BaseSlot
```

Comment:
Maybe we need some comments to describe the order logic.
StatPrepareSlot、RuleCheckSlot、StatSlot is the independent bucket, the ranking logic is independent.